api/src/feeds/impl/datasets_api_impl.py

```diff
@@ -95,10 +135,34 @@ def get_datasets_gtfs(query: Query, limit: int = None, offset: int = None) -> Li
             group_by=lambda x: x[0].stable_id,
         )
 
+        notices = DatasetsApiImpl._load_validation_report()
+
         gtfs_datasets = []
         for dataset_group in dataset_groups:
             dataset_objects, components, bound_box_strings, feed_ids = zip(*dataset_group)
             database_gtfs_dataset = dataset_objects[0]
+            notices_for_dataset = [notice for notice in notices if notice.dataset_id == database_gtfs_dataset.id]
+
+            validator_report = None
+            if notices_for_dataset:
```

Comment:
Let's add unit tests that cover this block of code.

Reply:
added unit tests for validation report